plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -79,4 +82,22 @@ public void testShowCreateTable()
                         format("   location = '.*/" + schemaName + "/region'\n") +
                         "\\)");
     }
+
+    @Test
+    public void testHiddenPathColumn()
+    {
+        String tableName = "hidden_file_column_table_metastore_" + randomTableSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a int, b VARCHAR) ");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, 'a')", 1);
+        assertQuery("SELECT a, b FROM " + tableName, "VALUES (1, 'a')");
+
+        MaterializedResult result = computeActual("SELECT file_path FROM \"" + tableName + "$files\"");
+        assertEquals(result.getRowCount(), 1);
+        String filePath = (String) result.getOnlyValue();
+
+        assertQuery("SELECT DISTINCT \"$path\" FROM " + tableName, format("VALUES " + "'" + filePath + "'"));
```

Comment:
Please add a another test logic with `WHERE \"$path\" = ...`. It's currently broken.

Reply:
Also, avoid mixing up `String.format` and `+` concat.

Reply:
@ebyhr added the `where "$path"` test as well.